### ISSUE TITLE
Checker reporting false errors for fluent style

### ISSUE THREAD

**Issue by msridhar** [kelloggm/typesafe-builder-checker#2]
We get false errors when code is written in a fluent style.  This change to `SimpleInference` shows the problem:

```diff
diff --git a/tests/basic/SimpleInference.java b/tests/basic/SimpleInference.java
index fff7da4..9ad1755 100644
--- a/tests/basic/SimpleInference.java
+++ b/tests/basic/SimpleInference.java
@@ -4,7 +4,7 @@ import org.checkerframework.checker.builder.qual.*;
 class SimpleInference {
     void build(@CalledMethods({"a"}) SimpleInference this) { }

-    void a() { }
+    SimpleInference a() { return this; }

     static void doStuffCorrect() {
         SimpleInference s = new SimpleInference();
@@ -12,6 +12,10 @@ class SimpleInference {
         s.build();
     }

+    static void doStuffCorrectFluent() {
+        (new SimpleInference()).a().build();
+    }
+
     static void doStuffWrong() {
         SimpleInference s = new SimpleInference();
         // :: error: method.invocation.invalid
```

**Comment by kelloggm** [comment]
#4 has an implementation for this.
